tools/wpt/browser.py

```diff
@@ -629,6 +628,44 @@ def version(self, binary=None, webdriver_binary=None):
         return match.group(1)
 
 
+class ChromeAndroid(ChromeAndroidBase):
+    """Chrome-specific interface for Android.
+    """
+
+    product = "chrome_android"
+    requirements = "requirements_chrome_android.txt"
+
+
+class AndroidWebview(ChromeAndroidBase):
+    """Webview-specific interface for Android.
+
+    Design doc:
+    https://docs.google.com/document/d/19cGz31lzCBdpbtSC92svXlhlhn68hrsVwSB7cfZt54o/view
+    """
+
+    product = "android_webview"
+    requirements = "requirements_chrome_android.txt"
+
+    def find_binary(self, venv_path=None, channel=None):
+        # Just get the current package name of the WebView provider.
+        # For WebView, it is not trivial to change the WebView provider, so
+        # we will just grab whatever is available.
+        # https://chromium.googlesource.com/chromium/src/+/HEAD/android_webview/docs/channels.md
+        command = ['adb', 'shell', 'dumpsys', 'webviewupdate']
+        try:
+            output = call(*command)
+        except (subprocess.CalledProcessError, OSError):
+            self.logger.warning("Failed to call %s" % " ".join(command))
+            return None
+        m = re.search(r'^\s*Current WebView package \(name, version\): \((.*), ([0-9.]*)\)$',
```

Comment:
Can we also use this for getting the version of WebView?

Reply:
Hmm... I don't think it's easy to retrofit it to the current logic of 1) getting the binary first and 2) get the version. And this separation allows us to provide a separate browser_binary, although I'm not sure if that's possible at all.